tests/beta/test_default_tensor.py

```diff
@@ -331,6 +338,304 @@ def test_hermitian_exceptions(self):
+        assert dev._nodes["state"][1].name == "BobCharlieState(1, 2)"
+        assert edges_valid(dev, num_nodes=2)
+
+
+        def test_add_initial_state_nodes_3_wires_beseparable_AC_B(self, tensornet_device_3_wires):
+            """Tests that biseparable AC|B initial states are properly created for a 3 wire device."""
+
+        dev = tensornet_device_3_wires
+        dev._clear_network_data()
+
+        tensors = [np.array([[1., 0.], [0., 1.]]) / np.sqrt(2), np.array([1., 1.]) / np.sqrt(2)]
+        wires = [[0, 2], [1]]
+        names = ["AliceCharlieState", "BobState"]
+        dev._add_initial_state_nodes(tensors, wires, names)
+
+        assert "state" in dev._nodes and len(dev._nodes) == 1
+        assert len(dev._nodes["state"]) == 2
+        assert dev._nodes["state"][0].name == "AliceCharlieState(0, 2)"
+        assert dev._nodes["state"][1].name == "BobState(1,)"
+        assert edges_valid(dev, num_nodes=2)
+
+
+        def test_add_initial_state_nodes_3_wires_tripartite_entangled(self, tensornet_device_3_wires):
+            """Tests that tripartite entangled initial states are properly created for a 3 wire device."""
+
+        dev = tensornet_device_3_wires
+        dev._clear_network_data()
+
+        tensors = [np.array([[1., 0., 0., 0.],
+                             [0., 0., 0., 0.],
+                             [0., 0., 0., 0.],
+                             [0., 0., 0., 1.]]) / np.sqrt(2)]
+        wires = [[0, 1, 2]]
+        names = ["GHZState"]
+        dev._add_initial_state_nodes(tensors, wires, names)
+
+        assert "state" in dev._nodes and len(dev._nodes) == 1
+        assert len(dev._nodes["state"]) == 1
+        assert dev._nodes["state"][0].name == "GHZState(0, 1, 2)"
+        assert edges_valid(dev, num_nodes=1)
+
+
+    @pytest.mark.parametrize("tensors,wires,names", [
+        ([np.array([[1., 0.], [0., 1.]]) / np.sqrt(2)], [[0,1]], ["A", "B"]),
+        ([np.array([[1., 0.], [0., 1.]]) / np.sqrt(2)], [[0], [1]], ["A"]),
+        ([np.array([1., 0.]), np.array([1., 1.]) / np.sqrt(2)], [[0]], ["A"]),
+    ])
+    def test_add_initial_state_nodes_exception(self, tensornet_device_2_wires, tensors, wires, names):
+        """Tests that an exception is given if the method _add_initial_state_nodes
+        receives arguments with incompatible lengths"""
+
+        dev = tensornet_device_2_wires
+        dev._clear_network_data()
+
+        with pytest.raises(ValueError, match="must all be the same length"):
+            dev._add_initial_state_nodes(tensors, wires, names)
+
+
+    def test_add_node(self, tensornet_device_2_wires, tol):
+        """Tests that the _add_node method functions correctly."""
+
+        dev = tensornet_device_2_wires
+
+        assert len(dev._nodes["state"]) == 2
+
+        zero_state = np.array([1., 0])
+        one_qubit_gate = np.array([[0, 1], [1, 0]])
+        two_qubit_gate = np.eye(4)
+        dev._add_node(one_qubit_gate, wires=[0], name="NewNodeX")
```

Comment:
Overall there seems to be a lot going on in this test, maybe a simpler version (e.g. only one node being added) could be extracted (followed by this one: multiple nodes which contain multi-qubit gates as well)?

Reply:
Thanks for suggestion, I split out the final part where it was checked if new keys were created. The earlier stuff I left in as one test (there are just a lot of attributes to check so it looks big, but I think it's ok this way and splitting more would be artificial)